remoteworkitem/trackerquery_repository.go

```diff
@@ -32,208 +30,128 @@ func NewTrackerQueryRepository(db *gorm.DB) *GormTrackerQueryRepository {
 
 // TrackerQueryRepository encapsulate storage & retrieval of tracker queries
 type TrackerQueryRepository interface {
-	CheckExists(ctx context.Context, id string) error
-	Create(ctx context.Context, query string, schedule string, tracker uuid.UUID, spaceID uuid.UUID) (*app.TrackerQuery, error)
-	Save(ctx context.Context, tq app.TrackerQuery) (*app.TrackerQuery, error)
-	Load(ctx context.Context, ID string) (*app.TrackerQuery, error)
-	Delete(ctx context.Context, ID string) error
-	List(ctx context.Context) ([]*app.TrackerQuery, error)
+	repository.Exister
+	Create(ctx context.Context, tq *TrackerQuery) error
+	Save(ctx context.Context, tq TrackerQuery) (*TrackerQuery, error)
+	Load(ctx context.Context, ID uuid.UUID) (*TrackerQuery, error)
```

Comment:
Why do we need to return a pointer here?

Reply:
To handle the case when trackerqury is not found